src/test/scala/org/broadinstitute/dsde/workbench/leonardo/auth/SamAuthProviderSpec.scala

```diff
@@ -0,0 +1,125 @@
+package org.broadinstitute.dsde.workbench.leonardo.auth
+
+import akka.http.scaladsl.testkit.ScalatestRouteTest
+import com.typesafe.config.Config
+import org.broadinstitute.dsde.workbench.google.mock.MockGoogleIamDAO
+import org.broadinstitute.dsde.workbench.leonardo.CommonTestData
+import org.broadinstitute.dsde.workbench.leonardo.dao.MockGoogleDataprocDAO
+import org.broadinstitute.dsde.workbench.leonardo.db.{DbSingleton, TestComponent}
+import org.broadinstitute.dsde.workbench.leonardo.model.NotebookClusterActions.{DeleteCluster, SyncDataToCluster}
+import org.broadinstitute.dsde.workbench.leonardo.model.ProjectActions.CreateClusters
+import org.broadinstitute.dsde.workbench.leonardo.model.{ClusterName, ServiceAccountProvider}
+import org.broadinstitute.dsde.workbench.leonardo.monitor.NoopActor
+import org.broadinstitute.dsde.workbench.leonardo.service.{LeonardoService, TestProxy}
+import org.broadinstitute.dsde.workbench.model.WorkbenchEmail
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+import org.scalatest.mockito.MockitoSugar
+import org.scalatest._
+import org.scalatest.concurrent.ScalaFutures
+
+class TestSamAuthProvider(authConfig: Config, serviceAccountProvider: ServiceAccountProvider) extends SamAuthProvider(authConfig, serviceAccountProvider) {
+  override val samClient = new MockSwaggerSamClient()
+}
+
+class SamAuthProviderSpec extends FreeSpec with ScalatestRouteTest with Matchers with MockitoSugar with BeforeAndAfter with BeforeAndAfterAll with TestComponent with CommonTestData with TestProxy with ScalaFutures with OptionValues {
+  val googleProject = project.value
+  val clusterName = name1.string
+
+  val routeTest = this
+
+  private val samAuthProvider = new TestSamAuthProvider(config.getConfig("auth.samAuthProviderConfig"),serviceAccountProvider)
+
+  val gdDAO = new MockGoogleDataprocDAO(dataprocConfig, proxyConfig, clusterDefaultsConfig)
+  val iamDAO = new MockGoogleIamDAO
+
+  val leo = new LeonardoService(dataprocConfig, clusterFilesConfig, clusterResourcesConfig, proxyConfig, swaggerConfig, gdDAO, iamDAO, DbSingleton.ref, system.actorOf(NoopActor.props), samAuthProvider, serviceAccountProvider, whitelist)
+
+  "should add and delete a notebook-cluster resource with correct actions for the user when a cluster is created and then destroyed" in isolatedDbTest {
+    samAuthProvider.samClient.billingProjects += (project, userInfo.userEmail) -> Set("launch_notebook_cluster")
+    // check the sam auth provider has no notebook-cluster resource
+    samAuthProvider.samClient.notebookClusters shouldBe empty
+    samAuthProvider.samClient.hasActionOnNotebookClusterResource(userInfo.userEmail, project, name1, "status") shouldBe false
```

Comment:
can I just make these assert(...) ?

Reply:
If you use `assert` you should `import org.scalatest.Assertions._` to make it use ScalaTest's `assert` instead of `scala.Predef.assert`. :)

But I would stick with the "should" matchers since that's our convention.